### PR TITLE
Add initial props

### PR DESCRIPTION
**Pull request by ferdinandsalis**
I am still quite lousy when it comes to writing tests … ☺️

### CODE REVIEW

**Comment by ferdinandsalis** [comment]
I am missing some things here. Disregard for now ☺️

File: .eslintrc [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "react/no-unused-prop-types": "off",
     "react/prefer-stateless-function": "off",
     "react/prop-types": ["error", { "skipUndeclared": true }],
+    "react/forbid-prop-types": "off",
```
Comment (MicheleBertoli) [inline .eslintrc]: What about `"react/forbid-prop-types": ["on", { "forbid": "any" }]`?

File: README.md [hunk 2]
```diff
@@ -100,8 +100,8 @@ exports[`it works: b 1`] = `
 
 ## withStatechart(statechart[, options])(Component)
 
-The `withStatechart` higher-order component takes a statechart definition (see [xstate](https://github.com/davidkpiano/xstate)), some optional [options](#options) and a component.
-It returns a new component with special [props](#props), [action methods](#action-methods) and [lifecycle methods](#lifecycle-methods).
```
Comment (MicheleBertoli) [inline README.md]: Can you please keep "methods" here?

File: src/withStatechart.js [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
-    }
-
     constructor(props) {
       super(props)
 
+      const { initial, initialData } = props
+
+      let initialMachineState
+      if (initial) {
+        initialMachineState = State.from(initial)
```
Comment (MicheleBertoli) [inline src/withStatechart.js]: `State.from` 👌

File: src/withStatechart.js [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
-      componentState: options.initialData,
-      machineState: this.machine.initialState,
-    }
-
     constructor(props) {
       super(props)
 
+      const { initial, initialData } = props
+
+      let initialMachineState
```
Comment (MicheleBertoli) [inline src/withStatechart.js]: nit: What about a ternary here?

File: README.md (hunk 2 above)
Comment (ferdinandsalis) [reply README.md]: Ups, that was by mistake.

File: src/withStatechart.js (hunk 4 above)
Comment (ferdinandsalis) [reply src/withStatechart.js]: Sure!

File: src/withStatechart.js (hunk 3 above)
Comment (ferdinandsalis) [reply src/withStatechart.js]: Yeah!

File: README.md [hunk 5]
```diff
@@ -100,8 +100,8 @@ exports[`it works: b 1`] = `
...
+The `withStatechart` higher-order component takes a statechart definition (see [xstate](https://github.com/davidkpiano/xstate)), some optional [options](#options) and a component. You can also set the `initial` machine state and `initialData` via props, e.g. `<withStatemachine(Component) initial="b" initialData={{ counter: 0 }} />`.
```
Comment (MicheleBertoli) [inline README.md]: I'd move `initialData` from options to props (and remove it from options) - I'm worried that having it in both places might be confusing.
Also, would you mind renaming `initial` to `initialMachineState`?
Thank you very much!

Comment (ferdinandsalis) [reply README.md]: Yes! I wanted to ask you that as well … will do

**Comment by ferdinandsalis** [comment]
Should have pushed this as a new commit for you to better track the changes. Sorry. Let me know if there are things you want done differently …

File: .eslintrc (hunk 1 above)
Comment (ferdinandsalis) [reply .eslintrc]: Yes. Sure. Missed that.

**Comment by MicheleBertoli** [comment]
This is awesome, thank you very much @ferdinandsalis.

The only missing bit is:
> Also, would you mind renaming `initial` to `initialMachineState`?

To give you more context: although `initial` is used in the definition of the machine, I have the feeling it might be confusing as a prop name.
What do you think?

I'm happy to merge and change the name later if you prefer.

**Comment by ferdinandsalis** [comment]
Yes I could see a problem with `initial`. What about just `initialState`, in `initialMachineState` Machine seems redundant when seen in context of usage `<StateMachine initialState={initialState} />`.

**Comment by MicheleBertoli** [comment]
Thank you very much, @ferdinandsalis.
Although `initialState` is cleaner, I'm worried it could be confused with the initial component state (naming is hard :)).

**Comment by MicheleBertoli** [comment]
Thanks to this change I realized the `initialData` option wasn't really needed (while the prop is), and I could use `defaultProps` :)

**Comment by ferdinandsalis** [comment]
I always forget about `defaultProps` ☺️

**Pull request merged**